src/rgw/rgw_sal_motr.cc

```diff
@@ -1770,20 +1754,11 @@ int MotrObject::MotrReadOp::prepare(optional_yield y, const DoutPrefixProvider*
 
   rgw_bucket_dir_entry ent;
   rc = source->get_bucket_dir_ent(dpp, ent);
-  if(ent.is_delete_marker())
-  {
-    rc = -ENOENT;
-    return rc;
-  }
   if (rc < 0)
     return rc;
 
-  // In case of un-versioned/suspended case,
-  // GET and HEAD object output should ignore versiodId field.
-  if (!source->have_instance()) {
```

Comment:
Hi @andriytk  we added this check for making MGW behavior inline with AWS & RADOS
short problem description: In case of null version object, GET/HEAD object should not display version-id field in response..
 Only case is when end user pass "--version-id null " as request parameters then only we should display null version-id as part of response.
 
please refer this JIRA : https://jts.seagate.com/browse/CORTX-31876

Reply:
It doesn't show it:

```
$ aws s3api get-object --endpoint-url http://localhost:8000 --bucket testbucket --key 8m /tmp/8m.check 
{
    "AcceptRanges": "bytes",
    "LastModified": "Mon, 20 Jun 2022 13:28:08 GMT",
    "ContentLength": 8388608,
    "ETag": "\"9da99fe0893981324aec37836e1f21cf\"",
    "ContentType": "binary/octet-stream",
    "Metadata": {}
}
$ aws s3api head-object --endpoint-url http://localhost:8000 --bucket testbucket --key 8m 
{
    "AcceptRanges": "bytes",
    "LastModified": "Mon, 20 Jun 2022 13:28:08 GMT",
    "ContentLength": 8388608,
    "ETag": "\"9da99fe0893981324aec37836e1f21cf\"",
    "ContentType": "binary/octet-stream",
    "Metadata": {}
}
```